middleware/auth.js

```diff
@@ -0,0 +1,37 @@
+const jwt = require('jsonwebtoken')
+const accessToken = require('../models/access_token');
+
+module.exports.auth = async (req, res, next) => {
+    try {
+        const token = req.headers.token
+        const decoded = jwt.verify(token, 'thisisprateek')
+        const user = await accessToken.findOne({ "user_id": decoded.id, "access_token": token })
+        if (!user) {
+            throw new Error("no access token or has expired or user is deleted")
+        }
```

Comment:
no need to check the expiry in the collection;
because this feature is execute by JWT itself